### DISCUSSION TITLE
Using binarySource=docker from within a docker container and baseDir

### DISCUSSION THREAD

**Discussion by Adirelle** [renovatebot/renovate#32964] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab -- 39.52.0

### Please tell us more about your question or problem

Hello,

When running tools using a sidecar container, renovate tries to share its `baseDir` using `--volume` so the tool can access the repository and the cache. However, when it is executed from inside a container in which the `baseDir` is mapped to a different path, e.g.:

```bash
docker run \
  --env RENOVATE_BASE_DIR=/data \
  --volume /path/to/volume:/data \
  --volume /var/run/docker.sock:/var/run/docker.sock \  
  renovate/renovate
```

It tries to run the sidecar using something like this:

```bash
docker run --volume /data:/data # rest of the sidecar parameters
```

This leads to mistakenly mount `/data` instead of `/path/to/volume` into the sidecar container.

A simple workaround is to use the same path for the `baseDir` inside the container:

```bash
docker run \
  --env RENOVATE_BASE_DIR=/path/to/volume \
  --volume /path/to/volume:/path/to/volume \
  --volume /var/run/docker.sock:/var/run/docker.sock \  
  renovate/renovate
```

I understand renovate inside the container has no way to get the external path of the `baseDir` to properly mount it when starting the sidecar container. However, I am wondering if this could be improved somehow:

* talk about this somewhere in the documentation ?
* trying to use `--volume-from` to copy the mounts from the original container ?
* introducing a configuration option for the sidecar ?

### Logs (if relevant)

--

**Comment by rarkins** [comment]
We prefer binarySource=install these days so haven't invested much time into docker sidecar in a while. You're certainly welcome to improve the documentation as a start. I thought we'd mentioned somewhere already about the fact volumes are just volumes and not internal
